contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -4057,3 +4064,317 @@ exec_stmt_fulltextindex(PLtsql_execstate *estate, PLtsql_stmt_fulltextindex *stm
+		}
+
+		/* check if the given number of boundaries are exceeding allowed limit */
+		if (nargs >= 15000)
+		{
+			ereport(ERROR, 
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("CREATE/ALTER partition function failed as only a maximum of 15000 partitions can be created.")));
+		}
+
+		values = palloc0(nargs * sizeof(Datum));
+
+		for (int i = 0; i < nargs; i++)
+		{
+			Datum val;
+
+			/* evaluate the value from the expr */
+			val = exec_eval_expr(estate, list_nth(arg, i), &isnull, &valtype, &valtypmod);
+
+			/* 
+			 * implicitly convert range values to specified parameter type
+			 * and raise error with ordinal position if conversion fails
+			 */
+			PG_TRY();
+			{
+				values[i] = exec_cast_value(estate, val, &isnull,
+								valtype, valtypmod,
+								typ->typoid, typ->atttypmod);
+			}
+			PG_CATCH();
+			{
+				MemoryContext err_ctx;
+				err_ctx = MemoryContextSwitchTo(cur_ctxt);
+				pfree(tsql_typename);
+				pfree(values);
+				FlushErrorState();
+				MemoryContextSwitchTo(err_ctx);
+				ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("Could not implicitly convert range values type specified at ordinal %d to partition function parameter type.",
+							i+1)));
+			}
+			PG_END_TRY();
+		}
+
+		/*
+		 * Find oid of comparator function for input type, which will be used during the sorting.
+		 * Here, we are first finding the default operator class for the input type then using that
+		 * we are finding the operator family for that operator class and finally using that we are
+		 * finding the defined comparator function for that operator family.
+		 */
+		basetype_oid = getBaseType(typ->typoid);
+		opclass_oid = GetDefaultOpClass(basetype_oid, BTREE_AM_OID);
+		opfamily_oid = get_opclass_family(opclass_oid);
+		cmpfunction_oid = get_opfamily_proc(opfamily_oid, basetype_oid, basetype_oid,
+							BTORDER_PROC);
+
+		/* set the function oid of operator in tsql comparator context */
+		cxt.function_oid = cmpfunction_oid;
+		cxt.colloid = tsql_get_server_collation_oid_internal(false);
+		cxt.contains_duplicate = false;
+
+		/* 
+		 * sort the datum values using quick sort, we don't need to worry about worst case
+		 * of quick sort here when the array is already sorted, the function qsort_arg()
+		 * itself first checks and returns the same array if values already sorted.
+		 */
+		qsort_arg(values, nargs, sizeof(Datum), tsql_compare_values, &cxt);
+
+		/* raise error if input contains duplicate value */
+		if (cxt.contains_duplicate)
+		{
+			ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("Duplicate values are not allowed in partition function boundary values list.")));
+		}
+
+		sql_variant_oid = (*common_utility_plugin_ptr->get_tsql_datatype_oid) ("sql_variant");
+		/* cast each value to sql_variant datatype */
+		for (int i = 0; i < nargs; i++)
+		{
+			values[i] = exec_cast_value(estate, values[i], &isnull,
```

Comment:
Are we overriding the previously allocated memory here?

Reply:
addressed it.